File: C++/Medium/Accepted/reduce-array-size-to-the-half_2.cpp
class Solution {
public:
    int minSetSize(vector<int>& arr) {
        
        map<int, int> m;
        int n = arr.size();
        for (int i = 0; i < n; i++) {
            m[arr[i]]++;
        }
        vector<int> v;
        for (auto it = m.begin(); it != m.end(); it++) {
            v.push_back(it->second);
        }
        sort(v.begin(), v.end(), greater<int>());
        int sum = 0;
        int count = 0;
        for (int i = 0; i < v.size(); i++) {
            sum += v[i];
            count++;
            if (sum >= n / 2) {
                break;
            }
        }
        return count;
    }
};

File: Java/Medium/Accepted/reduce-array-size-to-the-half_2.java
class Solution {
    public int minSetSize(int[] arr) {
        
        HashMap<Integer, Integer> map = new HashMap();
        for (int i = 0; i < arr.length; i++) {
            map.put(arr[i], map.getOrDefault(arr[i], 0) + 1);
        }
        PriorityQueue<Integer> heap = new PriorityQueue<>((a, b) -> b - a);
        for (int key : map.keySet()) {
            heap.offer(map.get(key));
        }
        int res = 0;
        int count = 0;
        while (count < arr.length / 2) {
            count += heap.poll();
            res++;
        }
        return res;
    }
}